azure/table_azure_resource.go

```diff
@@ -0,0 +1,330 @@
+			},
+
+			// Azure standard columns
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Location").Transform(toLower),
+			},
+			{
+				Name:        "resource_group",
+				Description: ColumnDescriptionResourceGroup,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ID").Transform(extractResourceGroupFromID),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listResources(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	session, err := GetNewSession(ctx, d, "MANAGEMENT")
+	if err != nil {
+		plugin.Logger(ctx).Error("azure_resource.listResources", "session_error", err)
+		return nil, err
+	}
+	subscriptionID := session.SubscriptionID
+
+	resourceClient := resources.NewClientWithBaseURI(session.ResourceManagerEndpoint, subscriptionID)
+	resourceClient.Authorizer = session.Authorizer
+
+	// https://learn.microsoft.com/en-us/rest/api/resources/resources/list?view=rest-resources-2021-04-01#uri-parameters
+	filter := getResourceFilter(d.Quals)
+	expand := "createdTime,changedTime,provisioningState"
+	// A value less than or equal to '1000' must be provided.
+	// Limiting the results
+	maxLimit := int32(1000)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			maxLimit = limit
+		}
+	}
+
+	result, err := resourceClient.List(ctx, filter, expand, &maxLimit)
+	if err != nil {
+		plugin.Logger(ctx).Error("azure_resource.listResources", "api_error", err)
+		return nil, err
+	}
+	for _, resource := range result.Values() {
+		d.StreamListItem(ctx, resource)
+		// Check if context has been cancelled or if the limit has been hit (if specified)
+		// if there is a limit, it will return the number of rows required to reach this limit
+		if d.RowsRemaining(ctx) == 0 {
+			return nil, nil
+		}
+	}
+
+	for result.NotDone() {
+		err = result.NextWithContext(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("azure_resource.listResources", "api_paging_error", err)
+			return nil, err
+		}
+		for _, resource := range result.Values() {
+			d.StreamListItem(ctx, resource)
+			// Check if context has been cancelled or if the limit has been hit (if specified)
+			// if there is a limit, it will return the number of rows required to reach this limit
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTION
+
+func getResource(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	session, err := GetNewSession(ctx, d, "MANAGEMENT")
+	if err != nil {
+		plugin.Logger(ctx).Error("azure_resource.getResource", "session_error", err)
+		return nil, err
+	}
+	subscriptionID := session.SubscriptionID
+	id := d.EqualsQuals["id"].GetStringValue()
+	if id == "" {
+		return nil, nil
+	}
+
+	apiVersion := "2021-04-01"
```

Comment:
@ParthaI is this the latest available API version?

Reply:
Yes, @misraved, it is the latest [API version](https://learn.microsoft.com/en-us/rest/api/resources/resources/get-by-id?view=rest-resources-2021-04-01#code-try-0). Also, it is required parameter to use the API Get By ID.